### DISCUSSION TITLE
blog/2022-06-13-just-use-multilevel-models-for-your-pre-post-rct-data/

### DISCUSSION THREAD

**Discussion by giscus** [ASKurz/blogdown5#13] [Comments]
# blog/2022-06-13-just-use-multilevel-models-for-your-pre-post-rct-data/

I've been thinking a lot about how to analyze pre/post control group designs, lately. Happily, others have thought a lot about this topic, too. The goal of this post is to introduce the change-score and ANCOVA models, introduce their multilevel-model counterparts, and compare their behavior in a couple quick simulation studies. Spoiler alert: The multilevel variant of the ANCOVA model is the winner.

https://solomonkurz.netlify.app/blog/2022-06-13-just-use-multilevel-models-for-your-pre-post-rct-data/

**Comment by andymilne** [comment]
Say you had a pre-test followed by two post-tests, so there are three time points. This would require coding the time variable as a factor with levels called something like "0", "1", and "2". But if time is a factor, the formula `1 + time + tx:time` now produces nonsensical terms in the design matrix; e.g., `time0:tx`.

Are you aware of any way around this problem?

**Reply by ASKurz** [reply]
You might also just treat time as continuous.

**Reply by andymilne** [reply]
In the case I am thinking of, that wouldn't be appropriate because there may well be a ceiling reached by time 1, so the improvement from time1 to time2 will be smaller than the difference from time1 to time2. I guess one way around it is to simply split the data into one set with time0 and time1, and one set with time1 and time2. But, it is nice to have everything in the same model, particularly if information can be shared between the three time points (e.g., participant level random intercepts). I do find the way that R adds in unwanted design matrix columns when using : for interactions very inconvenient and don't really understand the reason why it does that.

**Reply by ASKurz** [reply]
Perhaps you might code time as two dummy variables, instead of a factor. Assuming you had baseline as the reference category, your model syntax might be 

`y ~ 1 + t1 + t2 + tx:t1 + tx:t2 + (1 | id)`, 

where `t1` is the dummy for the second time point and `t2` is the dummy for the third time point. I don't love the way this model only allows for a random intercept, but so it goes.

**Reply by ASKurz** [reply]
But, if you weren't obsessed with a model that could make the plots like mine above, you could also just model the second two time points and use the baseline assessment as a covariate. That model could be something like 

`y ~ 1 + tx + time + tx:time + pre + (1 | id)`,

where `pre` is the baseline assessment. You could generalize this model in various ways. For ideas, check out the discussion in this recent twitter poll: https://twitter.com/SolomonKurz/status/1649139351522013197

**Reply by andymilne** [reply]
Both look like excellent ideas and I will experiment with them. At first sight, the first version seems preferable because the second version has t1 and t2 predicted by t0 (pre), but t2 is not predicted by t1, which seems inconsistent. 

Having said that, I generally (when there are only two time points) use `post ~ pre*tx` because it can naturally include a ceiling effect through a coefficient on `pre` that is smaller than 1 (as recommended Gelman and Hill in their 2007 book). You can also use a polynomial or smooth function of `pre` to incorporate floor and ceiling effects. That seems, I think, to be lost when predicting effects only with an indicator of time as in the `y ~ time + time:tx` approach. But I am now rambling, apologies ...

**Comment by trudy231** [comment]
Hi Solomon
thanks for this excellent post! One question regarding treating the variables as factor or as continous: What are your reasons for not treating your variables as factors? The reason why I'm asking is that when I follow your steps for the multi level approach, I get different results and also different outputs in R when treating time and/or treatment as continous vs. as factors. For the output, I would have expected a time, condition, and time*condition interaction component, but following your steps and treating the variables as continous, you only get a time and time*condition component. Any advice on this problem is very welcome. Thanks!!

**Reply by trudy231** [reply]
(Sorry, I didn't realize the asterisk made the text italic; it should be time x condition component).

**Reply by ASKurz** [reply]
Hey @trudy231, thanks for the kind words. I've played around with setting `time` as a factor, and I see what you mean. It's difficult to treat `time` as a factor and get the beta coefficients in the same metric as when you treat `time` as continuous. To answer your initial question, I have treated `time` as continuous because,  (a) to my mind that's the most natural thing to do, and (b) I already know it will produce unbiased estimates of the beta coefficients in the case of a simple study design. I imagine there is a way to recode `time` and or `tx` as factors to reproduce the parameterizations in the post, but I'm not sure what that would be ,and I'm not sure it's worth the effort to figure it out. But if you or anyone else does, I'd love to see it.

**Reply by trudy231** [reply]
Thanks again @ASKurz! Okay, then I'll stick with treating time as continuous. What about treatment? In your example, it is also treated as continuous, right?

**Reply by ASKurz** [reply]
In my example, `tx` is a dummy variable which is conceptually discrete, but technically continuous.

**Reply by trudy231** [reply]
Thanks again, @ASKurz!

**Reply by ASKurz** [reply]
Cheers!

**Comment by MarcosMorenoVerdu** [comment]
Hi Solomon,
Thanks a lot for your post, I find it really helpful!
I have a question related to the tx variable. In this case you coded it as a dummy variable with only two categories, control and treatment, and therefore the time:tx beta coefficient is the ATE. But what if tx has more than two categories? I have an RCT where 3 different treatments were compared against not receiving any intervention. Therefore my tx variable is coded 0 (control) and 1-3 for treatments A, B and C. Is then correct to run the multilevel model y ~ 1 + time + tx:time + (1 | id) with time and tx treated as continuous? It's difficult to interpret de beta of time:tx (at least to me). Should I at least treat tx as a factor in order to obtain specific betas for each level, and then interpret each beta as the ATE of each treatment? To complicate it more, I have also three time points (pre, post and follow-up)... Should I treat time as continuous or factor?

**Reply by ASKurz** [reply]
Hey @MarcosMorenoVerdu, thanks for the kind words. It appears your use case is much broader than the scope of this post, but I'll give this a shot. It seems like you have two kinds of questions, one with respect to multiple conditions, and the other with respect to multiple post-baseline time points. If you have 4 groups, the simplest way would be to include 3 dummy variables, one for each of the non-control conditions. In this way, the control condition would be the reference category. As to your case where there are three time points, this depends in part on what shape you expect the group means to follow over time. If you expect they will be linear across all three time points, I'd fit a linear growth model with random intercepts and linear slopes. However, if you expect some kind of non-linear trajectory for 1 or more of the groups, you're probably best just treating time as categorical and only setting the intercept to random.

**Reply by MarcosMorenoVerdu** [reply]
Hi Solomon,
Thanks for the quick response, your input is very valuable. Regarding the problem with the 4 groups, I did what you said. Just to tell you in case it might be interesting, that the betas were essentially the same for each tx when comparing the approach with 3 dummy variables, to the approach of using 1 categorical variable with 4 levels. The latter produced slightly greater standard errors, but not very relevant in practice.

With respect to the matter of how to model the effect of time, indeed I do not expect the response to follow a linear trajectory for the active treatment groups, so I'll go with the second option you comment. However I also tried the linear growth model approach and of course, the betas were different between them. However, in both cases, there were trivial differences between fitting the models with 3 dummy variables or with 1 categorical variable with 4 levels.

Thank you very much!!

**Comment by ASKurz** [comment]
@Lachlan-Cribb wrote a nice blog post extending this post by examining what happens when your baseline measures are truncated. He showed a nice solution is to allow the residual variances to vary between pre and post. You can find his post, *Baseline truncation in pre-post RCTs*, [here](https://lachlancribb.netlify.app/post/baseline-truncation-in-pre-post-rcts/).

**Reply by ASKurz** [reply]
Update: The link to @Lachlan-Cribb's post is now [here](https://lachlancribb.netlify.app/posts/baseline-truncation-in-rcts/index.en).

**Comment by ASKurz** [comment]
On a project for work, I had reason to work through how you might fit a multilevel version of the conventional ANOVA. To my mind, the simplest way to do this is to first add dummy-variable versions for `pre` and `post`. Once those are added to the long version of the data (i.e., `dl` in this post), here's how to fit the conventional and multilevel variants of the ANOVA model:

```{r}
# add pre and post dummies
dl <- dl %>% 
  mutate(pre = 1 - time,
         post = time)

# fit the conventional ANOVA
w0 <- glm(
  data = dw,
  family = gaussian,
  post ~ 1 + tx)

# fit the multilevel ANOVA
l0 <- lmer(
    data = dl,
    y ~ 0 + pre + pre:tx + post + post:tx + (1 | id))
```

In this context, you'll find the point estimate and its standard error for the multilevel `post:tx` term matches those for the `tx` term in the conventional ANOVA.

**Comment by jmk9186** [comment]
Hi Solomon, 

Thanks for this helpful post. There was one part I didn't quite follow...

I understand how w1 is a model of change scores and w2 is like an ANCOVA. 
w1 <- glm(
  data = dw,
  family = gaussian,
  (post - pre) ~ 1 + tx)

w2 <- glm(
  data = dw,
  family = gaussian,
  post ~ 1 + tx + pre)

But can you explain how l1 and l2 is the multilevel equivalents of w1 and w2, respectively? Ie. How is l1 a model of change scores? And how is l2 predicts post scores using baseline scores as a covariate?
l1 <- lmer(
  data = dl,
  y ~ 1 + tx + time + tx:time + (1 | id))

l2 <- lmer(
  data = dl,
  y ~ 1 + time + tx:time + (1 | id))

**Reply by ASKurz** [reply]
Hey @jmk9186. Thanks for your interest. 

Unfortunately, some of my wording in this post was a little limited. I wrote this before my deeper dive into change scores (see [here](https://solomonkurz.netlify.app/blog/2023-06-19-causal-inference-with-change-scores/)). During that deeper dive, I read O’Connell et al (https://doi.org/10.4172/2155-6180.1000334) who helped clarify how you can mix and match post and change-score models with ANOVAs and ANCOVAs. What I referred to as a multilevel change score model in this post, `y ~ 1 + tx + time + tx:time + (1 | id)`, might have been better described as a multilevel ANOVA. The most important point is that when you have a pre/post RCT, that model has less statistical power than the multilevel ANCOVA, `y ~ 1 + time + tx:time + (1 | id)`.

At some point, I might edit this post to better clarify the issue. If I do, I’ll reference your comment as an impetus.

**Comment by jschoeneberger** [comment]
@ASKurz, in the multilevel ANCOVA formulation, the intercept represents the adjusted pre-test value (i.e., the starting point from which the treatment effect is estimated, if you will).  Do you know, is there way to get that adjusted covariate value from the single-level model:
glm(formula = post ~ 1 + tx + pre, family = gaussian, data = dw)

Is there a way, from the glm formulation, to obtain the value of pre across the mean of the model (i.e., 0.11320 like is obtained as the intercept in the MLM ANCOVA)?

**Reply by ASKurz** [reply]
Hey @jschoeneberger, I think the only way you could do that is with some version of path analysis where you also have a model on `pre`, even if it's just an intercept. SEM software like **lavaan** or **blavaan** can do it, and you could also do it with **brms** using the multivariate syntax (see [here](https://cran.r-project.org/web/packages/brms/vignettes/brms_multivariate.html)).

**Comment by adrianolszewski** [comment]
In clinical trials we also commonly use the fixed-effect-only estimation for correlated observations: GLS (Generalized Least Square) and GEE (Generalised Estimating Equations). It has the advantage that we can precisely model the residual correlation when there are multiple timepoints (random intercept approximately corresponds to compound symmetry, which isn't something one would expect in longer studies...), so we can pick unstructured, ARx (e.g. AR1..n), ante-dependence, Topelitz, ..., and their heterogeneous forms. This in a combination with a specific form of the model: response ~ Timepoint * Treatment + Baseline_response + Baseline_response : Timepoint forms the popular MMRM specification (mixed-model for repeated measurements; "mixed" is a weird part here, as there are no random effects at all; it's rather a degenerated LMM with only the R-side covariance). In MMRM the response vector contains only the post-treatment observations, or change-from-baselines (in both cases needs to be adjusted for the baseline value; not a problem, leads to same treatment effect).

We can use it also for another very popular technique, the cLDA analysis (constrained longitudinal data analysis), where response contains both baseline and post-treatment measurements, but the baselines are forced to be equal across the arms (mA = mB) through a special form of the model: response ~ 0 + Treatment + Treatment : Timepoint (most resources I saw remove the intercept, some do not).

In R we can use either the old way: nlme::gls() or the modern one: mmrm package (GLS estimation). For GEE, which has the advantage of returning population-average estimates for the non-Gaussian GLM family, e.g. logistic regression for binary responses) we can use the glmtoolbox and geessbin for the Firth-type penalised GEE-logistic regression. All of them can be combined with the emmeans package, and most will work with the marginaleffects.

**Reply by adrianolszewski** [reply]
Forgot to add, that timepoint and treatment are both categorical (factors) in these settings.

**Reply by ASKurz** [reply]
Yep, I learned about the MMRM approach a couple years ago, and I liked what it had to offer. It's been on my blog topic to-do list ever since.
